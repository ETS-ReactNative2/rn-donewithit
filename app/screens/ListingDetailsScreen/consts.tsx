const CONSTS = {
  CONTAINER_PADDING_BOTTOM: 50,
  TITLE_FONT_SIZE: 25,
  SUBTITLE_FONT_SIZE: 18,
  PADDING_HORIZONTAL: 20,
  PADDING_VERTICAL: 15,
  LINE_HEIGHT: 40,
  IMAGE_HEIGHT: 300,
  AVATAR_SIZE: 80,
  TOTAL_LINE_HEIGHT: 30
}

export default CONSTS
