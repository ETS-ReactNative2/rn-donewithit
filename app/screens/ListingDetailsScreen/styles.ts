import { StyleSheet } from 'react-native'
import COLORS from '../../config/colors'
import CONSTS from './consts'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.BACKGROUND_COLOR
  },
  image: {
    width: '100%',
    height: CONSTS.IMAGE_HEIGHT
  },
  infoBox: {
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL,
    paddingVertical: CONSTS.PADDING_VERTICAL
  },
  text: {
    lineHeight: CONSTS.LINE_HEIGHT,
    fontWeight: 'bold'
  },
  title: {
    fontSize: CONSTS.TITLE_FONT_SIZE
  },
  subTitle: {
    fontSize: CONSTS.SUBTITLE_FONT_SIZE,
    color: COLORS.SECONDARY
  },
  sellerBox: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    marginTop: 36,
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL
  },
  avatar: {
    width: CONSTS.AVATAR_SIZE,
    height: CONSTS.AVATAR_SIZE,
    borderRadius: CONSTS.AVATAR_SIZE / 2,
    marginRight: CONSTS.PADDING_HORIZONTAL / 2
  },
  sellerText: {
    fontSize: CONSTS.SUBTITLE_FONT_SIZE
  },
  total: {
    color: COLORS.GRAY,
    lineHeight: CONSTS.TOTAL_LINE_HEIGHT
  }
})

export default styles
