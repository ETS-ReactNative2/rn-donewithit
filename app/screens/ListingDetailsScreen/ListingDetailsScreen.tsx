import { Image, View } from 'react-native'
import React, { useMemo } from 'react'
import AppText from '../../components/AppText'
import styles from './styles'
import { JACKET, MOSH } from '../../config/images'

export default function ListingDetailsScreen() {
  const { title, subTitle, image, avatarImage, name, listings } = useMemo(
    () => ({
      title: 'Red jacket for sale',
      subTitle: '$100',
      image: JACKET,
      avatarImage: MOSH,
      name: 'Mosh Hamedani',
      listings: 5
    }),
    []
  )

  return (
    <View style={styles.container}>
      <Image style={styles.image} source={image} />
      <View style={styles.infoBox}>
        <AppText style={[styles.text, styles.title]}>{title}</AppText>
        {subTitle && (
          <AppText style={[styles.text, styles.subTitle]}>{subTitle}</AppText>
        )}
      </View>
      <View style={styles.sellerBox}>
        <Image style={styles.avatar} source={avatarImage} />
        <View>
          <AppText style={styles.sellerText}>{name}</AppText>
          <AppText
            style={[styles.sellerText, styles.total]}
          >{`${listings} Listings`}</AppText>
        </View>
      </View>
    </View>
  )
}
