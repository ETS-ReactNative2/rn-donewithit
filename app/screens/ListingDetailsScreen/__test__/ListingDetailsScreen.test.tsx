import React from 'react'
import renderer from 'react-test-renderer'
import ListingDetailsScreen from '../index'

describe('ListingDetailsScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListingDetailsScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
