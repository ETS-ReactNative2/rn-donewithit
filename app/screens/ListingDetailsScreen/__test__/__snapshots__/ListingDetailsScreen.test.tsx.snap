// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f8f4f4",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 300,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "fontWeight": "bold",
              "lineHeight": 40,
            },
            Object {
              "fontSize": 25,
            },
          ],
        ]
      }
    >
      Red jacket for sale
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Courier",
            "fontSize": 13,
          },
          Array [
            Object {
              "fontWeight": "bold",
              "lineHeight": 40,
            },
            Object {
              "color": "turquoise",
              "fontSize": 18,
            },
          ],
        ]
      }
    >
      $100
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "marginTop": 36,
        "paddingHorizontal": 20,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 40,
          "height": 80,
          "marginRight": 10,
          "width": 80,
        }
      }
    />
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Courier",
              "fontSize": 13,
            },
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Mosh Hamedani
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Courier",
              "fontSize": 13,
            },
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#6e6969",
                "lineHeight": 30,
              },
            ],
          ]
        }
      >
        5 Listings
      </Text>
    </View>
  </View>
</View>
`;
